core/jobs_test.py

```diff
@@ -85,6 +90,42 @@ def test_create_new(self):
         self.assertFalse(MockJobManagerOne.is_active(job_id))
         self.assertFalse(MockJobManagerOne.has_finished(job_id))
 
+    def test_base_job_manager_enqueue_raises_error(self):
+        with self.assertRaisesRegexp(
+            NotImplementedError,
+            'Subclasses of BaseJobManager should implement _real_enqueue().'):
+            jobs.BaseJobManager._real_enqueue(  # pylint: disable=protected-access
+                'job_id', taskqueue_services.QUEUE_NAME_DEFAULT, None)
+
+    def test_failing_jobs(self):
+        observed_log_messages = []
+
+        def _mock_logging_function(msg, *args):
+            """Mocks logging.error()."""
+            observed_log_messages.append(msg % args)
+
+        logging_swap = self.swap(logging, 'error', _mock_logging_function)
+
+        # Mocks GoogleCloudStorageInputReader() to fail a job.
+        _mock_input_reader = lambda _, __: 1 / 0
+
+        input_reader_swap = self.swap(
+            input_readers, 'GoogleCloudStorageInputReader', _mock_input_reader)
+        assert_raises_context_manager = self.assertRaises(Exception)
+
+        job_id = MockJobManagerOne.create_new()
+        store_map_reduce_results = jobs.StoreMapReduceResults()
+
+        with input_reader_swap, assert_raises_context_manager, logging_swap:
+            store_map_reduce_results.run(
+                job_id, 'core.jobs_test.MockJobManagerOne', 'output')
+
+        expected_log_message = 'Job %s failed at' % job_id
+
+        self.assertEqual(len(observed_log_messages), 2)
+        self.assertTrue(
```

Comment:
Add a comment to explain why you are ignoring log message 0.

In general, if a reviewer needs to ask you about something, that might suggest that a comment would be useful (for future readers of the code).

Reply:
done